site/com_joomgallery/src/Model/UsercategoryModel.php

```diff
@@ -0,0 +1,229 @@
+   *
+   * @param   integer   $id  The id of the primary key.
+   *
+   * @return  bool when root category is matching id
+   *
+   * @throws \Exception
+   * @since   4.2.0
+   */
+  public function isUserRootCategory(int $id): bool
+  {
+    $isUserRootCategory = false;
+
+    try
+    {
+      $db = Factory::getContainer()->get(DatabaseInterface::class);
+
+      // Check number of records in tables
+      $query = $db->createQuery()
+        ->select('id, parent_id')
+        ->from($db->quoteName(_JOOM_TABLE_CATEGORIES))
+        ->where($db->quoteName('id') . ' = ' . (int) $id);
+
+      $db->setQuery($query);
+      $item = $db->loadObject();
+
+      if((!empty($item->id)) && $item->parent_id == 1)
+      {
+        $isUserRootCategory = true;
+      }
+    }
+    catch(\RuntimeException $e)
+    {
+      Factory::getApplication()->enqueueMessage('isUserRootCategory-Error: ' . $e->getMessage(), 'error');
+
+      return false;
+    }
+
+    return $isUserRootCategory;
+  }
+
+  /**
+   * Method to get the profile form.
+   *
+   * The base form is loaded from XML
+   *
+   * @param   array   $data      An optional array of data for the form to interogate.
+   * @param   bool    $loadData  True if the form is to load its own data (default case), false if not.
+   *
+   * @return  Form|CurrentUserInterface|false    A Form object on success, false on failure
+   *
+   * @throws \Exception
+   * @since   4.2.0
+   */
+  public function getForm($data = [], $loadData = true): Form|CurrentUserInterface|false
+  {
+    // Get the form.
+    $form = $this->loadForm($this->typeAlias, 'usercategory', ['control' => 'jform', 'load_data' => $loadData]);
+
+    if(empty($form))
+    {
+      return false;
+    }
+
+    return $form;
+  }
+
+  /**
+   * Method to get the data that should be injected in the form.
+   *
+   * @return  \Joomla\CMS\Object\CMSObject|\stdClass|array  The default data is an empty array.
+   *
+   * @since   4.2.0
+   */
+  protected function loadFormData(): \Joomla\CMS\Object\CMSObject|\stdClass|array
+  {
+    return parent::loadFormData();
+  }
+
+  /**
+   * Get the return URL.
+   *
+   * @return  string  The return URL.
+   *
+   * @since   4.2.0
+   */
+  public function getReturnPage(): string
+  {
+    return base64_encode($this->getState('return_page', ''));
+  }
+
+  /**
+   * Get the most recent error message.
+   * Eiter generated in model or written to component
+   *
+   * @param   integer   $i  Option error index.
+   * @param   boolean   $toString  Indicates if Exception objects should return their error message.
+   *
+   * @return  string   Error message
+   *
+   * @since   1.7.0
+   *
+   * @deprecated  3.1.4 will be removed in 6.0
+   *              Will be removed without replacement
+   *              Catch thrown Exceptions instead of getError
+   */
+  public function getError($i = null, $toString = true)
+  {
+    // 2025.10.04: user images checkin does not find error message in model !manuel!
```

Comment:
getError und setError soll man sowieso nicht mehr verwenden.
Deshalb habe ich die `$this->component->setError(...)`. Diese wird es auch nach der deprecation durch Joomla noch geben.